### ISSUE TITLE
yq --inplace changes the last file given, not the first

### ISSUE THREAD

**Issue by parmentelat** [mikefarah/yq#1193]
the documentation in this page
<https://mikefarah.gitbook.io/yq/commands/evaluate-all#examples>
says that the `--inplace` option will change the first yaml file given

my observations show otherwise, here it's the second file, so I presume it's the last one

```bash
/tmp #
yq --version
yq (https://github.com/mikefarah/yq/) version 4.24.5
/tmp #
cat > a.yaml
a: 1
/tmp #
cat > b.yaml
b: 2
/tmp #
ls -l a.yaml b.yaml
-rw-r--r-- 1 root root 5 Apr 22 15:35 a.yaml
-rw-r--r-- 1 root root 5 Apr 22 15:35 b.yaml
/tmp #
yq ea a.yaml b.yaml
a: 1
---
b: 2
/tmp #
yq --inplace ea a.yaml b.yaml
/tmp #
ls -l a.yaml b.yaml
-rw-r--r-- 1 root root  5 Apr 22 15:35 a.yaml
-rw-r--r-- 1 root root 14 Apr 22 15:35 b.yaml
/tmp #
```

**Comment by mikefarah** [comment]
Yep, I can replicate this! It looks like it happens when no expression is passed in. Will fix.

**Comment by parmentelat** [comment]
FTR my actual use case is with using `--from-file` - so here again no expression passed on the command line, maybe that's a more accurate description ?

in any case please make sure to advertise if the behaviour changes as it will then break my current code ;)

thanks !

**Comment by mikefarah** [comment]
Fixed in 4.25.1 - the first file is now updated correctly!

**Comment by parmentelat** [comment]
thanks !
